pagopa/io-app
# chore: [IOPID-882] MP initialization refactoring

## Short description
This PR changes Mixpanel initialization flow according to the provided JIRA task [IOPID-882](https://pagopa.atlassian.net/browse/IOPID-882).

## List of changes proposed in this pull request
- Split initialization from identification.
- Use of reset function during a logout or wrong session.

## How to test
Run the app against the dev server, and check that all use cases provided [here](https://pagopa.atlassian.net/wiki/spaces/CM1/pages/692224316/Proposta+di+remediation+al+tracciamento+degli+eventi+Mixpanel+prima+dell+opt-in+da+parte+dell+utente#Sintesi-del-protocollo-di-Identity-managment) are covered. 
Beware that if you have opted out, the `mixpanel` instance is not instantiated.

<details><summary>Patch to apply logs</summary>

```diff
diff --git a/ts/mixpanel.ts b/ts/mixpanel.ts
index 8f5b8adee..8ffb48b9f 100644
--- a/ts/mixpanel.ts
+++ b/ts/mixpanel.ts
@@ -24,6 +24,7 @@ export const initializeMixPanel = async () => {
   const privateInstance = new MixpanelInstance(mixpanelToken);
   await privateInstance.initialize();
   mixpanel = privateInstance;
+  console.log(">>>>>>>>> INITIALIZE mixpanel " + JSON.stringify(mixpanel));
   // On app first open
   // On profile page, when user opt-in
   await setupMixpanel(mixpanel);
@@ -53,16 +54,22 @@ const setupMixpanel = async (mp: MixpanelInstance) => {
 
 export const identifyMixpanel = async () => {
   // Identify the user using the device uniqueId
+  console.log(
+    ">>>>>>>>> IDENTIFY (profile) mixpanel " + JSON.stringify(mixpanel)
+  );
   await mixpanel?.identify(getDeviceId());
 };
 
 export const resetMixpanel = async () => {
   // Reset mixpanel auto generated uniqueId
+  console.log(">>>>>>>>> RESET mixpanel " + JSON.stringify(mixpanel));
   await mixpanel?.reset();
 };
 
 export const terminateMixpanel = async () => {
+  console.log(">>>>>>>>> OPT-OUT (onboarding) " + JSON.stringify(mixpanel));
   if (mixpanel) {
+    console.log(">>>>>>>>> OPT-OUT (profile) " + JSON.stringify(mixpanel));
     await mixpanel.flush();
     await mixpanel.optOutTracking();
     mixpanel = undefined;
```
<p>



</p>
</details> 


[IOPID-882]: https://pagopa.atlassian.net/browse/IOPID-882?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
